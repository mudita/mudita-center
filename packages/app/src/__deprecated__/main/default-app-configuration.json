{ "osVersion": "1.1.0", "centerVersion": "1.5.0" }
