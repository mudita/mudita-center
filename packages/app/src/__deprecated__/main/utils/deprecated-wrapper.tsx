/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import { FunctionComponent } from "react"

type Element = FunctionComponent<any>

const deprecated = (component: Element, message?: string): Element => {
  if (process.env.NODE_ENV === "development") {
    const basicMessage = `This component is deprecated.`
    if (message) {
      console.warn(`${basicMessage} ${message}`)
    } else {
      console.warn(basicMessage)
    }
  }

  return component
}

export default deprecated
