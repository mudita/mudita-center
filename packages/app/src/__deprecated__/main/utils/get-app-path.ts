/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import path from "path"
import packageInfo from "../../../../package.json"

export enum AppType {
  Main = "main",
  Renderer = "renderer",
}
export type AppResolver = () => {
  type: AppType
  app: { getPath: (arg0: string) => string }
}

export const resolve: AppResolver = () => {
  // AUTO DISABLED - fix me if you like :)
  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
  const isRenderer = require("is-electron-renderer")
  // FIXME: Using remote module in renderer process isn't a good pattern.
  //  You can read more in https://github.com/electron/remote#migrating-from-remote
  // AUTO DISABLED - fix me if you like :)
  // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
  const app = isRenderer
    ? // AUTO DISABLED - fix me if you like :)
      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
      require("@electron/remote").app
    : // AUTO DISABLED - fix me if you like :)
      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
      require("electron").app
  const type = isRenderer ? AppType.Renderer : AppType.Main
  return {
    // AUTO DISABLED - fix me if you like :)
    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
    app,
    type,
  }
}

const getAppPath = (): string => {
  try {
    return path.join(resolve().app.getPath("appData"), packageInfo.name)
  } catch {
    return ""
  }
}

export default getAppPath
