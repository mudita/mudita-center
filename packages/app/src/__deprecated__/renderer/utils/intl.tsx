/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import React from "react"
import { createIntl } from "react-intl"
import translationConfig from "App/translations.config.json"
import localeEn from "App/__deprecated__/renderer/locales/default/en-US.json"
import extractLanguageKeys from "App/__deprecated__/renderer/utils/extract-test-locale"

const testLocale = extractLanguageKeys(localeEn)

export const intl = createIntl({
  locale: translationConfig.defaultLanguage,
  messages: process.env.NODE_ENV === "test" ? testLocale : localeEn,
})

// https://github.com/formatjs/formatjs/issues/1467#issuecomment-543872950
let index = 0
export const textFormatters = {
  b: (str: string) => <strong key={`bold-${++index}`}>{str}</strong>,
  u: (str: string) => <u key={`underline-${++index}`}>{str}</u>,
  i: (str: string) => <em key={`italics-${++index}`}>{str}</em>,
  s: (str: string) => <s key={`strike-${++index}`}>{str}</s>,
}
