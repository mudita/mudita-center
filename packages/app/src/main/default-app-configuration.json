{ "osVersion": "0.72.1", "centerVersion": "16.0.0" }
