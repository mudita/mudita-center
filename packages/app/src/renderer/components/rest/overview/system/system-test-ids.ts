/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/LICENSE.md
 */

export enum SystemTestIds {
  DownloadButton = "download-button",
}
