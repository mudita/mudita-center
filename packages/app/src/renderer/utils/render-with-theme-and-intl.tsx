/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import {
  Queries,
  render,
  RenderOptions,
  RenderResult,
} from "@testing-library/react"
import React from "react"
import { IntlProvider } from "react-intl"
import translationConfig from "App/translations.config.json"
import localeEn from "Renderer/locales/default/en-US.json"
import { ThemeProvider } from "styled-components"
import theme from "../styles/theming/theme"
import extractLanguageKeys from "Renderer/utils/extract-test-locale"
import ReactModal from "react-modal"

ReactModal.setAppElement(document.createElement("div"))
const testLocale = extractLanguageKeys(localeEn)

export function renderWithThemeAndIntl(
  ui: React.ReactElement,
  options?: Omit<RenderOptions, "queries">
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options: RenderOptions<Q>
): RenderResult
export function renderWithThemeAndIntl<Q extends Queries>(
  ui: React.ReactElement,
  options?: RenderOptions<Q>
): RenderResult {
  return render(
    <ThemeProvider theme={theme}>
      <IntlProvider
        defaultLocale={translationConfig.defaultLanguage}
        locale={translationConfig.defaultLanguage}
        messages={process.env.NODE_ENV === "test" ? testLocale : localeEn}
      >
        {ui}
      </IntlProvider>
    </ThemeProvider>,
    // @ts-ignore
    options
  )
}
