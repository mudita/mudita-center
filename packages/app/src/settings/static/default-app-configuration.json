{
  "centerVersion": "2.0.2",
  "productVersions": {
    "MuditaPure": "1.6.0",
    "MuditaHarmony": "2.0.0"
  }
}
