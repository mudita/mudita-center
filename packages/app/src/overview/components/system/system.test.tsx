/**
 * Copyright (c) Mudita sp. z o.o. All rights reserved.
 * For licensing, see https://github.com/mudita/mudita-center/blob/master/LICENSE.md
 */

import "@testing-library/jest-dom/extend-expect"
import React, { ComponentProps } from "react"
import { fireEvent } from "@testing-library/dom"
import { renderWithThemeAndIntl } from "Renderer/utils/render-with-theme-and-intl"
import { intl } from "Renderer/utils/intl"
import System from "App/overview/components/system/system.component"
import { SystemTestIds } from "App/overview/components/system/system-test-ids.enum"
import { osVersion } from "App/main/default-app-configuration.json"

type Props = ComponentProps<typeof System>

const defaultProps: Props = {
  osVersion,
}

const render = (extraProps?: Partial<Props>) => {
  const props = {
    ...defaultProps,
    ...extraProps,
  }
  return renderWithThemeAndIntl(<System {...props} />)
}

test("matches snapshot", () => {
  const { container } = render()
  expect(container).toMatchSnapshot()
})

test("renders os version properly", () => {
  const { getByTestId } = render()
  expect(getByTestId(SystemTestIds.OsVersion)).toHaveTextContent(
    `[value] module.overview.muditaOsUpdateTitle ${osVersion}`
  )
})

test("renders available update info properly", () => {
  const { getByText } = render({ updateAvailable: true })
  expect(
    getByText(
      intl.formatMessage({ id: "module.overview.systemUpdateAvailable" })
    )
  ).toBeInTheDocument()
})

test("renders You're up to date info properly", () => {
  const { getByText } = render({ updateAvailable: false })
  expect(
    getByText(
      intl.formatMessage({ id: "module.overview.systemUpdateUpToDate" })
    )
  ).toBeInTheDocument()
})

test("renders 'check for updates' button properly", () => {
  const { queryByRole } = render()
  expect(queryByRole("button")).toHaveTextContent(
    intl.formatMessage({ id: "module.overview.systemCheckForUpdates" })
  )
})

test("renders 'update now' button properly", () => {
  const { queryByRole } = render({ updateAvailable: true })
  expect(queryByRole("button")).toHaveTextContent(
    intl.formatMessage({ id: "module.overview.systemDownloadAction" })
  )
})

test("checks for update after button click", () => {
  const onUpdateCheck = jest.fn()

  const { getByRole } = render({
    onUpdateCheck,
  })

  fireEvent.click(getByRole("button"))

  expect(onUpdateCheck).toHaveBeenCalled()
})

test("triggers download after button click", () => {
  const onDownload = jest.fn()

  const { getByRole } = render({
    updateAvailable: true,
    onDownload,
  })

  fireEvent.click(getByRole("button"))

  expect(onDownload).toHaveBeenCalled()
})

test("triggers update after button click", () => {
  const onUpdate = jest.fn()

  const { getByRole } = render({
    updateAvailable: true,
    updateDownloaded: true,
    onUpdate,
  })

  fireEvent.click(getByRole("button"))

  expect(onUpdate).toHaveBeenCalled()
})
