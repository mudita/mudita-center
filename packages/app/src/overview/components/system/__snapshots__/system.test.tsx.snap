// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
.c1 {
  margin: 0;
  font-size: 2.4rem;
  line-height: 0.8;
  font-weight: 700;
  -webkit-letter-spacing: -0.06rem;
  -moz-letter-spacing: -0.06rem;
  -ms-letter-spacing: -0.06rem;
  letter-spacing: -0.06rem;
}

.c5 {
  margin: 0;
  font-size: 1.4rem;
  line-height: 1.05;
  color: #6a6a6a;
  font-weight: 300;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
}

.c6 {
  margin: 0;
  font-size: 1.6rem;
  line-height: 1.05;
}

.c7 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  color: #6a6a6a;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
}

.c14 {
  margin: 0;
  font-size: 1.2rem;
  line-height: 1.2;
  text-transform: uppercase;
  -webkit-letter-spacing: 0.1rem;
  -moz-letter-spacing: 0.1rem;
  -ms-letter-spacing: 0.1rem;
  letter-spacing: 0.1rem;
}

.c11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: none;
  -webkit-transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  transition: background 250ms ease-in-out, color 250ms ease-in-out, border 250ms ease-in-out;
  cursor: pointer;
  outline: none;
  box-sizing: border-box;
  width: 17rem;
  height: 4rem;
  color: #ffffff;
  border-radius: 0.5rem;
  background: #3b3f42;
  border: 0.1rem solid #3b3f42;
}

.c11:hover {
  background: #232426;
}

.c11 g {
  fill: #ffffff;
}

.c15 {
  white-space: nowrap;
}

.c9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  position: relative;
}

.c12 {
  --radius: 0.5rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  z-index: 1;
  pointer-events: all;
  z-index: 3;
}

.c12:disabled {
  pointer-events: none;
}

.c12:hover {
  z-index: 2;
}

.c12:not(:first-of-type) {
  border-radius: 0;
  margin-left: -0.1rem;
}

.c12:first-of-type:not(:last-of-type) {
  border-radius: var(--radius) 0 0 var(--radius);
}

.c12:last-of-type:not(:first-of-type) {
  border-radius: 0 var(--radius) var(--radius) 0;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  padding: 4rem;
  box-sizing: border-box;
  border-radius: 0.5rem;
  background-color: #ffffff;
}

.c3 {
  grid-area: Text;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-end;
  -webkit-box-align: flex-end;
  -ms-flex-align: flex-end;
  align-items: flex-end;
  margin-top: auto;
}

.c10 {
  grid-area: Buttons;
  justify-self: end;
  min-width: 18rem;
}

.c13 {
  padding: 0 1.6rem;
  width: 50%;
}

.c13 svg {
  height: initial;
  width: initial;
}

.c4 span {
  margin-left: 0.8rem;
  font-weight: 300;
  text-transform: none;
}

.c4 p {
  margin-bottom: 0.8rem;
}

.c8 {
  margin-top: 0.8rem;
  text-transform: none;
  display: -webkit-inline-box;
  display: inline-box;
  padding: 0.3rem 0.5rem;
  border-radius: 0.4rem;
  -webkit-letter-spacing: 0.04rem;
  -moz-letter-spacing: 0.04rem;
  -ms-letter-spacing: 0.04rem;
  letter-spacing: 0.04rem;
  background-color: #f4f5f6;
}

<div>
  <section
    class="c0"
  >
    <div>
      <h2
        class="c1"
      >
        [value] module.overview.muditaOsUpdateTitle
      </h2>
    </div>
    <div
      class="c2"
    >
      <div
        class="c3"
      >
        <div
          class="c4"
        >
          <p
            class="c5"
          >
            Current version:
          </p>
          <p
            class="c6"
            data-testid="os-version"
          >
            [value] module.overview.muditaOsUpdateTitle
             0.72.1
          </p>
        </div>
        <p
          class="c7 c8"
        >
          [value] module.overview.systemUpdateUpToDate
        </p>
      </div>
      <section
        class="c9 c10"
      >
        <button
          class="c11 c12 c13"
          size="2"
          type="button"
        >
          <p
            class="c14 c15"
          >
            [value] module.overview.systemCheckForUpdates
          </p>
        </button>
      </section>
    </div>
  </section>
</div>
`;
